realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -112,431 +112,304 @@ public TableQuery not() {
-        return this;
-    }
-
-    public TableQuery lessThan(long[] columnKey, long[] tablePtrs, long value) {
-        nativeLess(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery lessThanOrEqual(long[] columnKey, long[] tablePtrs, long value) {
-        nativeLessEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery between(long[] columnKey, long value1, long value2) {
-        nativeBetween(nativePtr, columnKey, value1, value2);
-        queryValidated = false;
-        return this;
-    }
-
-    // Queries for float values.
-
-    public TableQuery equalTo(long[] columnKey, long[] tablePtrs, float value) {
-        nativeEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery notEqualTo(long[] columnKey, long[] tablePtrs, float value) {
-        nativeNotEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery greaterThan(long[] columnKey, long[] tablePtrs, float value) {
-        nativeGreater(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery greaterThanOrEqual(long[] columnKey, long[] tablePtrs, float value) {
-        nativeGreaterEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery lessThan(long[] columnKey, long[] tablePtrs, float value) {
-        nativeLess(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
-
-    public TableQuery lessThanOrEqual(long[] columnKey, long[] tablePtrs, float value) {
-        nativeLessEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
+    public static String buildSortDescriptor(String[] fieldNames, Sort[] sortOrders) {
+        StringBuilder descriptorBuilder = new StringBuilder("SORT(");
 
-    public TableQuery between(long[] columnKey, float value1, float value2) {
-        nativeBetween(nativePtr, columnKey, value1, value2);
-        queryValidated = false;
-        return this;
-    }
+        String sortSeparator = "";
+        for (int i = 0; i < fieldNames.length; i++) {
+            String fieldName = fieldNames[i];
 
-    // Queries for double values.
+            descriptorBuilder.append(sortSeparator)
+                    .append(escapeFieldName(fieldName))
+                    .append(" ")
+                    .append((sortOrders[i] == Sort.ASCENDING) ? "ASC" : "DESC");
 
-    public TableQuery equalTo(long[] columnKey, long[] tablePtrs, double value) {
-        nativeEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
+            sortSeparator = ", ";
```

Comment:
This seems to lead to trailing `, `. That is okay with the Query Parser?

Reply:
It is fine! We use the separator between fields.